tests/FunctionsTest.php

```diff
@@ -93,6 +99,38 @@ public function testUsesPhpCompilationOfJsonIfPossible()
         $this->assertSame($soughtData, Aws\load_compiled_json($jsonPath));
     }
 
+    /**
+     * @covers Aws\load_compiled_json()
+     */
+    public function testOnlyLoadsCompiledJsonOnce()
+    {
+        $soughtData = ['foo' => 'bar'];
+        $jsonPath = sys_get_temp_dir() . '/some-file-name-' . time() . '.json';
+
+        file_put_contents($jsonPath, json_encode($soughtData), LOCK_EX);
+
+        $this->assertSame($soughtData, Aws\load_compiled_json($jsonPath));
+        $jsonAtime = fileatime($jsonPath);
+
+        file_put_contents($jsonPath, 'INVALID JSON', LOCK_EX);
+
+        $compiledPath = "{$jsonPath}.php";
+        file_put_contents(
+            $compiledPath,
+            '<?php return ' . var_export($soughtData, true) . ';',
+            LOCK_EX
+        );
+
+        $this->assertSame($soughtData, Aws\load_compiled_json($jsonPath));
+        $compiledAtime = fileatime($compiledPath);
+
+        sleep(1);
```

Comment:
Maybe use [usleep](https://www.php.net/manual/en/function.usleep.php) to take less time?

Reply:
fileatime() returns seconds, so we need a full second to guarantee there would be difference in the access times on the test.